pkg/scheduler/internal/cache/cache.go

```diff
@@ -401,13 +402,33 @@ func (cache *schedulerCache) GetPod(pod *v1.Pod) (*v1.Pod, error) {
 	return podState.pod, nil
 }
 
+func isNodeReady(node *v1.Node) bool {
+	if node == nil {
+		return false
+	}
+	for _, cond := range node.Status.Conditions {
+		if cond.Type == v1.NodeReady {
+			return cond.Status == v1.ConditionTrue
+		}
+	}
+	return false
+}
+
 func (cache *schedulerCache) AddNode(node *v1.Node) error {
 	cache.mu.Lock()
 	defer cache.mu.Unlock()
 
 	n, ok := cache.nodes[node.Name]
 	if !ok {
 		n = schedulernodeinfo.NewNodeInfo()
+		if utilfeature.DefaultFeatureGate.Enabled(features.TaintNodesByCondition) && !isNodeReady(node) {
+			n.SetTaints([]v1.Taint{
```

Comment:
Are we persisting this into the cache ? What happens when the `nodeInformer` calls the updateFunc `updateNodeInCache(oldNode,newNode)` ? 

I'm not sure this will solve the problem.

Reply:
This solves first half of the problem when new nodes are added, when the nodes are updated, the problem will still be there(although we can have the same logic of check if the node is ready or not in update fn too) . I think @Huang-Wei is suggesting one possible path where the changes could be made.